src/Payments/PreMountLoader.res

```diff
@@ -1,165 +1,129 @@
+      let customerPaymentMethodsPromise = PaymentHelpers.fetchCustomerPaymentMethodList(
         ~clientSecret,
         ~publishableKey,
         ~optLogger=Some(logger),
         ~switchToCustomPod=false,
         ~endpoint,
       )
-    | _ => JSON.Encode.null->Promise.resolve
-    }
 
-    let sessionTokensResponse = switch hyperComponentName {
-    | Elements =>
-      PaymentHelpers.fetchSessions(
+      let sessionTokensPromise = PaymentHelpers.fetchSessions(
         ~clientSecret,
         ~publishableKey,
         ~optLogger=Some(logger),
         ~switchToCustomPod=false,
         ~endpoint,
         ~merchantHostname,
       )
-    | _ => JSON.Encode.null->Promise.resolve
-    }
 
-    let savedPaymentMethodsResponse = switch hyperComponentName {
-    | PaymentMethodsManagementElements =>
-      PaymentHelpers.fetchSavedPaymentMethodList(
+      let messageHandler = (ev: Window.event) => {
+        let json = ev.data->Utils.safeParse
+        let dict = json->Utils.getDictFromJson
+        if dict->Dict.get("sendPaymentMethodsResponse")->Option.isSome {
+          paymentMethodsPromise->sendPromiseData("payment_methods")
+        } else if dict->Dict.get("sendCustomerPaymentMethodsResponse")->Option.isSome {
+          customerPaymentMethodsPromise->sendPromiseData("customer_payment_methods")
+        } else if dict->Dict.get("sendSessionTokensResponse")->Option.isSome {
+          sessionTokensPromise->sendPromiseData("session_tokens")
+        }
+      }
+
+      let promises = [paymentMethodsPromise, customerPaymentMethodsPromise, sessionTokensPromise]
+      (promises, messageHandler)
+    })
+
+    React.null
+  }
+}
+
+module PreMountLoaderForPMMElements = {
+  @react.component
+  let make = (~logger, ~endpoint, ~ephemeralKey) => {
+    useMessageHandler(() => {
+      let savedPaymentMethodsPromise = PaymentHelpers.fetchSavedPaymentMethodList(
         ~ephemeralKey,
         ~optLogger=Some(logger),
         ~switchToCustomPod=false,
         ~endpoint,
       )
-    | _ => JSON.Encode.null->Promise.resolve
-    }
-
-    (
-      paymentMethodsResponse,
-      customerPaymentMethodsResponse,
-      sessionTokensResponse,
-      savedPaymentMethodsResponse,
-    )
-  })
 
-  let sendPromiseData = (promise, key) => {
-    open Promise
-    promise
-    ->then(res => {
-      messageParentWindow([("response", res), ("data", key->JSON.Encode.string)])
-      switch key {
-      | "payment_methods" => setPaymentMethodsResponseSent(_ => true)
-      | "session_tokens" => setSessionTokensResponseSent(_ => true)
-      | "customer_payment_methods" => setCustomerPaymentMethodsResponseSent(_ => true)
-      | "saved_payment_methods" => setSavedPaymentMethodsResponseSent(_ => true)
-      | _ => ()
+      let messageHandler = (ev: Window.event) => {
+        let json = ev.data->Utils.safeParse
+        let dict = json->Utils.getDictFromJson
+        if dict->Dict.get("sendSavedPaymentMethodsResponse")->Belt.Option.isSome {
```

Comment:
```suggestion
        if dict->Dict.get("sendSavedPaymentMethodsResponse")->Option.isSome {
```

Reply:
Can we have 
```
let isKeyPresentInDict = (dict, key) => dict->Dict.get(key)->Option.isSome
```

this in Utils and use that everywhere as this is been called multiple times